docs/use/toolchain.md

```diff
@@ -0,0 +1,81 @@
+# Lind toolchain
+
+The toolchain to build and run lind programs consists of the following components:
+
+- *Clang* with *WebAssembly System Interface* (WASI) support to build *glibc*
+  and lind programs
+- A custom [*glibc*](../internal/libc.md) used as sysroot to build
+  lind programs
+- A custom `wasm-opt` binary to enable [multi-processing](../internal/multiprocess-support.md)
+  in lind programs
+- A custom *WebAssembly* runtime ([`wasmtime`](../internal/wasmtime.md)) based
+  on [*RawPOSIX*](../internal/rawposix.md) to run lind programs
+- Rust Cargo to build `wasmtime`
+
+This document gives an overview of how the toolchain is built. The build process
+is automated with *Docker*, *Bazel* and custom shell scripts. Please refer to
+the relevant files linked below for details about the build commands and used
+options.
+
+
+## Building the toolchain step by step
+
+1. __Install system dependencies__ *(see `apt` in [Dockerfile](https://github.com/Lind-Project/lind-wasm/blob/main/.devcontainer/Dockerfile))*
+
+2. __Download *Clang* and install builtins__ *(see `wget` and `cp` in [Dockerfile](https://github.com/Lind-Project/lind-wasm/blob/main/.devcontainer/Dockerfile))*
+
+    *Clang* supports the *WASI* cross-compilation target out of
+    the box, provided the necessary compiler runtime
+    [builtins](https://clang.llvm.org/docs/Toolchain.html#compiler-rt-llvm) and
+    a matching [sysroot](https://clang.llvm.org/docs/CrossCompilation.html).
+    See [*wasi-sdk* docs](https://github.com/WebAssembly/wasi-sdk) for details.
+
+    A pre-built *Clang* can be downloaded from the
+    [llvm-project releases](https://github.com/llvm/llvm-project/releases/) page.
+    Matching builtins are available in the *lind-wasm* repo under
+    [`src/glibc/wasi`](https://github.com/Lind-Project/lind-wasm/tree/main/src/glibc/wasi).
+
+3. __Build *glibc* and generate sysroot__ (see `make_glibc` in [BUILD](https://github.com/Lind-Project/lind-wasm/blob/main/BUILD) file)
+    1. Configure and compile *glibc* for the *WASI* target with *Clang*.  (see
+      [`wasm-config.sh`](https://github.com/Lind-Project/lind-wasm/blob/main/src/glibc/wasm-config.sh))
```

Comment:
Let's adds some information about the used build flags here, like the current [build glibc doc](https://github.com/Lind-Project/lind-wasm/blame/997a29c89f1d26e400f7f18eca062914f78c62c9/docs/build/build-libc.md#L132-L145) does, but with better explanation of why these flags are are used with the given values.

Most notably, why do we use ["i686-linux-gnu" as `--host` and `--build` argument](https://github.com/Lind-Project/lind-wasm/blob/997a29c89f1d26e400f7f18eca062914f78c62c9/src/glibc/wasm-config.sh#L8)?

Reply:
@qianxichen233